src/protocols/abstract/Base.ts

```diff
@@ -1,20 +1,46 @@
 'use strict';
 
-const KuzzleError = require('../../KuzzleError');
-const { uuidv4 } = require('../../utils/uuidv4');
-const { KuzzleEventEmitter } = require('../../core/KuzzleEventEmitter');
-const PendingRequest = require('./PendingRequest');
-
-class KuzzleAbstractProtocol extends KuzzleEventEmitter {
-  constructor (host, options = {}, name = undefined) {
+import { KuzzleError } from '../../KuzzleError';
+import { uuidv4 } from '../../utils/uuidv4';
+import { KuzzleEventEmitter } from '../../core/KuzzleEventEmitter';
+import { PendingRequest } from './PendingRequest';
+import { KuzzleRequest, JSONObject } from '../../utils/interfaces';
+
+/**
+ * @todo proper TS conversion
+ */
+export abstract class KuzzleAbstractProtocol extends KuzzleEventEmitter {
+  private _pendingRequests: Map<string, PendingRequest>;
+  private _host: string;
+  private _name: string;
+  private _port: number;
+  private _ssl: boolean;
+
+  public id: string;
+
+  public state: string;
+
+  constructor (host: string, options: JSONObject = {}, name: string = undefined) {
     super();
 
     this._pendingRequests = new Map();
     this._host = host;
     this._name = name;
     const port = parseInt(options.port, 10);
     this._port = isNaN(port) ? 7512 : port;
-    this._ssl = typeof options.sslConnection === 'boolean' ? options.sslConnection : false;
+
+    if (typeof options.ssl === 'boolean') {
```

Comment:
(nitpicking) we should probably add an assertion check verifying if either ssl OR sslConnection is set, but not both